### PR TITLE
Use _path_component_blobs attribute for dataset blobs

### PR DESCRIPTION
**Pull request by convoliution**
On top of #776

Use a standard interface for accessing the inner component blobs.

This avoids requiring
```python
if S3:
    blob._msg.s3.components
elif Path:
    blob._msg.path.components
else:
    raise TypeError
```
whenever a caller wants to access the components.

### CODE REVIEW

File: client/verta/tests/test_versioning/test_dataset.py [hunk 1]
```diff
@@ -9,9 +9,9 @@ def test_s3_bucket(self):
         pytest.importorskip("boto3")
 
         dataset = verta.dataset.S3("s3://verta-starter")
-        assert len(dataset._msg.s3.components) > 1
```
Comment (convoliution) [inline client/verta/tests/test_versioning/test_dataset.py:12]: Skip this file when reviewing; it's just a find-and-replace.

**Review by conradoverta — APPROVED** [review]
Nice!

File: client/verta/tests/test_versioning/test_dataset.py (hunk 1 above)
Comment (convoliution) [reply client/verta/tests/test_versioning/test_dataset.py:12]: Unfortunately, it's a bit more complex than a find-and-replace for the S3 tests now, but it's fairly trivial and they all still pass.

**Comment by convoliution** [comment]
@conradoverta Re-requesting review because I realized `S3DatasetBlob`, compared to `PathDatasetBlob`, has its `PathDatasetComponentBlob` nested one level deeper.

**Review by conradoverta — APPROVED** [review]

**Pull request merged**